### DISCUSSION TITLE
Almost 6 hours

### DISCUSSION THREAD

**Discussion by arcanosam** [leejet/stable-diffusion.cpp#430] [Q&A] [answered]
Hi all.

I have an Acer A515-52G-78HE and it's have a GEFORCE MX 130.

I used the sd-master-e410aeb-bin-win-avx2-x64 binary because using CPU-Z I saw my processor has AVX2 instruction.

I tryed before the vulkan binary option, but it didn't work. I don't know why anymore... well,

I just want to share that it took almost six hours to process the example command in README.md and it worked perfect, generating the same image.

But my questions are:

- cudart-sd-bin-win-cu12-x64.zip : this DLLs must be used? I put in the same directory as the sd executable, but I don't know if it is necessary
- There is something I could do to improve this process time? (or only buying another machine with a decent video card?  😅)

Below, my command prompt screen.

Thanks and great job with this project

![2024-10-04_stable_diffusion_cpp_init](https://github.com/user-attachments/assets/7e05c4a0-4980-4df3-bb08-2bc6246509f9)

![2024-10-04_stable_diffusion_cpp_finish](https://github.com/user-attachments/assets/e09e7901-0b0b-4d3f-9569-e53c88346acb)

**Comment by stduhpf** [comment, marked as answer]
Vulkan won't work if you can't fit the whole model + latent image buffer in video memory, so it's not surprising it refused to work on your laptop. 

SD3 is a fairly big model compared to others like SD 1.5 and SDXL, the diffusion model alone is over 4 GB in f16, and with the text encoders it's over 16GB.  One thing you could try to impove the generation time is to use a quantized version of the model. This will definitely reduce its memory footprint and maybe improve generaion time a bit (don't expect miracles).

**Comment by Green-Sky** [comment]
There are a couple of things you can try:
- quantizing the diffusion model to q8_0. This should keep the quality basically the same, but almost half the file size.
- use a model that converges faster like hyper or lcm etc.

**Comment by arcanosam** [comment]
Thank you both @Green-Sky @stduhpf 

I'm very noob on LLM for generative images, so I looked for chatgpt to give me some clarifications on what are in your answers.

So, I could try just change the model parameter to something like https://huggingface.co/second-state/stable-diffusion-3-medium-GGUF/blob/main/sd3-medium-Q8_0.gguf, right?

Is there some quantized option to FLUX.dev?

**Comment by arcanosam** [comment]
Actually I think I found both of my answers on README.md.

I can use for flux this link: https://huggingface.co/leejet/FLUX.1-dev-gguf?show_file_info=flux1-dev-q8_0.gguf

and in the https://github.com/leejet/stable-diffusion.cpp/blob/master/docs/flux.md#convert-flux-weights there is an example using gguf file as model

Thanks once more.

**Comment by mchome** [comment]
Use quantized lcm with TAESD in cudart or just buy new gear instead of torturing your potato.

**Comment by arcanosam** [comment]
Thanks @mchome 😅

I try only with the parameter which reduce the steps and the final image was very low resolution.

Indeed it all points that I have to buy another machine.

Anyone knows how is the suppor for those new Microsoft Copilot+ brand? [New Copilot+ PCs from the biggest brands available...]( https://blogs.microsoft.com/blog/2024/05/20/introducing-copilot-pcs/) stands more in the end of the article...

**Reply by stduhpf** [reply]
I don't think stable-diffusion.cpp supports the "Copilot+" NPUs. (I might be wrong). 
What you should be looking for is a computer with a (fast) GPU at least 12 GB of VRAM. If that's out of your budget, the next best thing is a computer with lots of RAM and a lot of fast CPU cores. (I've heard that Macs with "M" chips works pretty well, il you can bear with the ecosystem)
